### DISCUSSION TITLE
Consider using syslog for logging

### DISCUSSION THREAD

**Discussion by aleksandr-vin** [Lennolium/swiftGuard#7] [Feature Requests]
# 👉 Describe the problem

Reinventing the logging system will take time and effort which should be spent on implementing distinguishing features instead.

# 👥 Problem evidence & reach

1. `swiftguard` is a daemon process, which should constantly run in the background. Log files should be rotated, by size and potentially by time.

# 🥰 Describe the "impact" on users?

Taking all free space on the user home can lead to problems with further running the system.

# 🏆 How to solve this problem

Use standard ways to log events, like syslog. It is rotated by default. Plus power users can set it up separately to send logs to other syslog servers if they maintain their system logs on another server.

**Comment by aleksandr-vin** [comment]
Plus Console.app can be used to view the logs.

**Comment by Lennolium** [comment]
Good idea. Never thought about using a python package for it, but like it is most of the time better to use a well implemented package instead of trying to implement it by yourself. I will take a look into syslog for next commit 😄

**Comment by Lennolium** [comment]
I implemented the standard python logging module into the project. Its working like a charm, right now I am in the process of reorganizing the project structure (files, folders ...) to match the de facto standard in the python world 😸

I can't express how much I've to thank you for pushing me in the right direction! I did work the whole day on logging, but it was worth my time.

Cheers
